db-engine/src/main/java/com/splicemachine/db/impl/sql/conn/GenericLanguageConnectionContext.java

```diff
@@ -3986,6 +3987,15 @@ public void logEndFetching(String uuid, String statement, long fetchedRows) {
         }
     }
 
+    String getStatmentId(String stmt)
+    {
+        try {
+            return SQLStatement.getSqlStatement(stmt).getId();
+        } catch (Exception e) {
+            return "X#ERROR#";
+        }
+    }
+
```

Comment:
This should be constant for any prepared statement, is it possible to compute it during query compilation only once? If we do it early enough it can also be reused by DB-11957